src/models/jira/IssueTable.gs

```diff
@@ -0,0 +1,207 @@
+// Node required code block
+const IssueTableStorage_ = require('../IssueTableStorage.gs').IssueTableStorage_;
+// End of Node required code block
+
+/* ######## DEV- WIP - Testing #################### */
+
+
+function saveTest() {
+  var table1 = new IssueTable_();
+  table1.data.tableId = 'aaa';
+  
+  var table2 = new IssueTable_();
+  table2.data.tableId = 'bbb';
+  
+  var tblIndex = new IssueTableIndex_();
+  tblIndex.addTable(table1).addTable(table2);
+}
+
+function loadTest() {
+  console.log('current sheet id: %s', sheetIdPropertySave());
+  var tblIndex = new IssueTableIndex_();
+  var tableOld = tblIndex.getTable('bbb');
+  console.log('tableOld B: %s', tableOld.toJson());
+  
+  var tblIndex = new IssueTableIndex_();
+  var tableOld = tblIndex.getTable('aaa');
+  console.log('tableOld A: %s', tableOld.toJson());
+}
+
+
+
+var TestTable = {
+  sheetId: '6.02713257E8',
+  /*
+  Range names:
+  - Can contain only letters, numbers, and underscores.
+  - Can't start with a number, or the words "true" or "false."
+  - Can't contain any spaces or punctuation.
+  - Must be 1–250 characters.
+  - Can't be in either A1 or R1C1 syntax. For example, you might get an error if you give your range a name like "A1:B2" or "R1C1:R2C2."
+  */
+  rangeName: 'table1_6_02713257E8',
+  rangeA1: 'B3:D6',
+  headerRowOffset: 1,
+  headerValues: ['colA', 'colB', 'colC']
+};
+
+function testDefineRange() {
+  var ss = SpreadsheetApp.getActiveSpreadsheet();
+  var range = getTicketSheet().getRange(TestTable.rangeA1);
+  ss.setNamedRange(TestTable.rangeName, range);
+  
+  // check
+  var rangeCheck = ss.getRangeByName(TestTable.rangeName);
+  var rangeCheckName = rangeCheck.getA1Notation();
+  console.log('rangeCheckName: %s === TestTable.rangeA1', rangeCheckName, TestTable.rangeA1);
+}
+
+function testTable1() {
+  console.log('testTable1()');
+
+  SpreadsheetTriggers_.register('onEdit', 'onEditTableMeta', true);
+  SpreadsheetTriggers_.register('onChange', 'onEditTableMeta', true);
+}
+
+
+function onEditTableMeta(e) {
```

Comment:
I think we should put this function in Code.gs so all the Google sheet event handlers are in one place. That could just call a function in this file then.

Reply:
The functions really long already - its going to be hard to read once you get all the functionality in.

Reply:
WIP - its part of my prototyping, not the final function yet...